### PR TITLE
Added support of token classification to PredictionDrift & LabelDrift

### PR DESCRIPTION
**Pull request by nirhutnik**

### CODE REVIEW

File: deepchecks/nlp/checks/model_evaluation/prediction_drift.py [hunk 1]
```diff
@@ -143,28 +144,31 @@ def run_logic(self, context: Context) -> CheckResult:
...
-        if proba_drift:
-            if context.is_multi_label_task():
-                raise DeepchecksValueError('Cannot use proba drift mode for multi-label tasks')
-            train_prediction = np.array(model.predict_proba(train_dataset))
-            test_prediction = np.array(model.predict_proba(test_dataset))
-        elif context.is_multi_label_task():
-            train_prediction = convert_multi_label_to_multi_class(model.predict(train_dataset), model_classes)
-            test_prediction = convert_multi_label_to_multi_class(model.predict(test_dataset), model_classes)
+        if context.task_type == TaskType.TOKEN_CLASSIFICATION:
+            train_prediction = np.array(list(itertools.chain(*model.predict(train_dataset)))).reshape(-1, 1)
```
Comment (Nadav-Barak) [inline deepchecks/nlp/checks/model_evaluation/prediction_drift.py:152]: i think prediction are np arrays (need to verify for token classification) so the itertools may be redundant

**Review by Nadav-Barak — COMMENTED** [review]
Clean solution
a. verify it works with nones
b. verify the checks display looks ok

File: deepchecks/nlp/checks/model_evaluation/prediction_drift.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nirhutnik) [reply deepchecks/nlp/checks/model_evaluation/prediction_drift.py:152]: Of course it does, our whole functionality works with nones...

File: deepchecks/nlp/checks/model_evaluation/prediction_drift.py [hunk 2]
```diff
@@ -143,28 +144,31 @@ def run_logic(self, context: Context) -> CheckResult:
...
+            proba_drift = False
         else:
-            train_prediction = np.array(model.predict(train_dataset)).reshape((-1, 1))
-            test_prediction = np.array(model.predict(test_dataset)).reshape((-1, 1))
+            # Flag for computing drift on the probabilities rather than the predicted labels
+            proba_drift = ((len(context.model_classes) == 2) and (self.drift_mode == 'auto')) or \
+                          (self.drift_mode == 'proba')
+
+            if proba_drift:
+                if context.is_multi_label_task():
```
[the remark this answers is not in the document]
Comment (Nadav-Barak) [reply deepchecks/nlp/checks/model_evaluation/prediction_drift.py:161]: raise error if the user selected proba drift on a task of token classification

File: tests/nlp/checks/model_evaluation/prediction_drift_test.py [hunk 3]
```diff
@@ -95,3 +95,33 @@ def test_just_dance_small_drift(just_dance_train_test_textdata_sampled):
...
+
+    # Assert
+    assert_that(result.value['Drift score'], close_to(0, 0.01))
+
+
+def test_token_classification_with_nones(small_wikiann_train_test_text_data):
+    # Arrange
+    train, test = small_wikiann_train_test_text_data
+    train_label = train.label
+    train_label[0][0] = None
```
Comment (Nadav-Barak) [inline tests/nlp/checks/model_evaluation/prediction_drift_test.py:117]: you got confused in the tests

File: deepchecks/nlp/checks/model_evaluation/prediction_drift.py (hunk 1 above)
Comment (Nadav-Barak) [reply deepchecks/nlp/checks/model_evaluation/prediction_drift.py]: do the same for label drift :)

File: tests/nlp/checks/model_evaluation/prediction_drift_test.py (hunk 3 above)
Comment (nirhutnik) [reply tests/nlp/checks/model_evaluation/prediction_drift_test.py:117]: Nope, didn't :) note that I take the label and pass it as the prediction as I have no model predictions for this test dataset

**Review by Nadav-Barak — APPROVED** [review]

File: deepchecks/nlp/checks/model_evaluation/prediction_drift.py (hunk 1 above)
Comment (nirhutnik) [reply deepchecks/nlp/checks/model_evaluation/prediction_drift.py]: OOps

File: deepchecks/nlp/checks/model_evaluation/prediction_drift.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nirhutnik) [reply deepchecks/nlp/checks/model_evaluation/prediction_drift.py]: Agreed to raise warnings
